docs/user-stories.md

```diff
@@ -0,0 +1,111 @@
+# GLVD User Stories
+
+# USER-01
+
+As a user of Garden Linux I want to know about security issues in Garden Linux so I can operate my systems responsibly.
+
+## Acceptance Criteria 
+
+- [ ] The user can query for known CVEs of a list packages
+
+## Additional details
+
+- This does not yet require a nice user interface, an HTTP API is sufficient
+- This does not yet include knowledge about which packages are included in any given Garden Linux image, the user provides a list of package names and versions
+
+# USER-02
+
+As a user of Garden Linux I want to subscribe to updates on new security issues in the packages I care about.
+
+## Acceptance Criteria 
+
+- [ ] The user can get notifications for new known CVEs affecting their list of packages
+- [ ] The user can unsubscribe from notifications
+
+## Additional details
+
+- This might be implemented in an anonymous way where the user gets a tracking url/rss feed they need to bookmark
+- The backend/db needs to keep track of tracking urls
+- Users might want to subscribe with their email address; this introduces a new class of issues legally (data protection)
+
+# USER-03
+
+As a user of Garden Linux I want to track security issues in any given Garden Linux *image* so I can make an informed decision about needed actions in my system.
+
+## Acceptance Criteria 
+
+- [ ] The user can query for Garden Linux images such as `gardenlinux-gcp-gardener-prod-amd64-1443-5-bfb687a7`
+
+
+# USER-04
+
+As a user of Garden Linux I want to track security issues that additional packages bring into my image.
+
+## Acceptance Criteria 
+
+- [ ] The user can provide additional packages from the Garden Linux APT repo (in the same release as the image in use)
+
+# USER-05
+
+As a user of Garden Linux I want to be notified of new CVEs in the Garden Linux patch version I'm using.
+
+## Acceptance Criteria 
+
+- [ ] I can select the Garden Linux version (like 1443.5) I'm using to be notified on new CVEs for that specific version
+- [ ] I get a notification for new patch releases in that version (like 1443.6) so I can upgrade as soon as possible
+- [ ] I can select a Garden Linux major version (like 1443) to be notified on new and fixed CVEs in that version
+
+# USER-06
+
+As a user of Garden Linux I want to be notified of new major versions of Garden Linux so I can plan to upgrade.
+
+# USER-07
+
+As a user of GLVD I want a simple web interface to work with the service in a convenient way.
+
+## Acceptance Criteria 
+
+- [ ] Features described in the previous user stories can be accessed in a minimal web interface
+
+# USER-08
+
+As a user of GLVD I want a CLI tool to interact with the service.
+
+## Acceptance Criteria 
+
+- [ ] Features described in the previous user stories can be accessed using a command line tool
+
+# USER-09
+
+As a user of Garden Linux (container images) I want to be informed about additional issues installed packages bring.
```

Comment:
I might be too much to ask for, but I would *love* to have a moth type of message in case a given release becomes vulnerable. 

Ubuntu does this in a more gravy-ish way, by including it with the `/etc/update-motd.d/` for the landscape tool. It aims to be more of a selling point, but the core of the idea would be that we can generate such type of message for the user to be informed that the current running installation is vulnerable. 

Please tell me if I want overboard here.

Reply:
Sounds reasonable to me, and I'm not aware of any challenges with this yet.. This would mean that an http request needs to be done each time the motd is shown/updated, maybe caching would be good there and maybe this should be an optional feature.. but I like the idea.

Regarding the Gardener use-case of GL I think this is not super common that people login via ssh and will see the motd, but maybe we should also look into more observation tooling such as Prometheus? Just thinking about that now.